### ISSUE TITLE
Latest TensorFlow fails to install due to yesterday's Keras requirement change 

### ISSUE THREAD

**Issue by wei-v-wang** [tensorflow/tensorflow#52289]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version: latest as of https://github.com/tensorflow/tensorflow/commit/ca0fa154128d793df5e6809cb86336ac1fed3466 
- Python version: python3.8
- Installed using virtualenv? pip? conda?: pip 
- Bazel version (if compiling from source):  3.7.2
- GCC/Compiler version (if compiling from source): GCC9.3
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the problem**

**Build was successful, but installation of the wheel failed with the following Error message
09:34:20  ERROR: No matching distribution found for keras<2.8,>=2.7 (from tensorflow==2.8.0)**

**Provide the exact sequence of commands / steps that you executed before running into the problem**

pip install /tmp/pip3/tensorflow-2.8.0-cp38-cp38-linux_x86_64.whl

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

You can take a detailed look here: https://tensorflow-ci.intel.com/job/tensorflow-mkl-build-whl-nightly/941/console

**Comment by elfringham** [comment]
Broken for Linaro CI as well.
https://ci.linaro.org/job/ldcg-python-manylinux-tensorflow-nightly/123/console
The subtype: label for Ubuntu seems wrong, this is not related to the OS in use.

**Comment by sachinprasadhs** [comment]
Could please provide the link or document of wheel file which you are trying.Thanks.

**Comment by wei-v-wang** [comment]
@sachinprasadhs , this is a link that shows we were able to build TensorFlow successfully. But failed at installation steps. 
https://tensorflow-ci.intel.com/job/tensorflow-mkl-build-whl-nightly/942/console

02:46:49  [0m[91mINFO: Elapsed time: 2622.058s, Critical Path: 429.44s
02:46:49  INFO: 11295 processes: 1302 internal, 9993 local.
02:46:49  [0m[91mINFO: Build completed successfully, 11295 total actions
02:46:49  [0m[91mINFO: Build completed successfully, 11295 total actions
02:46:49  [0mFri Oct 8 09:46:36 UTC 2021 : === Preparing sources in dir: /tmp/tmp.2qdIZr4JRR
02:46:49  /tensorflow /tensorflow
02:47:01  /tensorflow
02:47:01  /tensorflow/bazel-bin/tensorflow/tools/pip_package/build_pip_package.runfiles/org_tensorflow /tensorflow
02:47:01  /tensorflow
02:47:01  /tmp/tmp.2qdIZr4JRR/tensorflow/include /tensorflow
02:47:01  /tensorflow
02:47:01  Fri Oct 8 09:47:00 UTC 2021 : === Building wheel
02:47:01  [91mwarning: no files found matching 'README'
02:47:01  [0m[91mwarning: no files found matching '*.pyd' under directory '*'
02:47:02  [0m[91mwarning: no files found matching '*.pyi' under directory '*'
02:47:02  [0m[91mwarning: no files found matching '*.pd' under directory '*'
02:47:03  [0m[91mwarning: no files found matching '*.dylib' under directory '*'
02:47:04  [0m[91mwarning: no files found matching '*.dll' under directory '*'
02:47:04  [0m[91mwarning: no files found matching '*.lib' under directory '*'
02:47:04  [0m[91mwarning: no files found matching '*.csv' under directory '*'
02:47:04  warning: no files found matching '*.h' under directory 'tensorflow/include/tensorflow'
02:47:04  [0m[91mwarning: no files found matching '*.proto' under directory 'tensorflow/include/tensorflow'
02:47:04  [0m[91mwarning: no files found matching '*' under directory 'tensorflow/include/third_party'
02:47:51  [0mFri Oct 8 09:47:45 UTC 2021 : === Output wheel file is in: /tmp/pip3
02:47:51  Processing /tmp/pip3/tensorflow-2.8.0-cp38-cp38-linux_x86_64.whl
02:47:51  Requirement already satisfied, skipping upgrade: wheel<1.0,>=0.32.0 in /usr/lib/python3/dist-packages (from tensorflow==2.8.0) (0.34.2)
02:47:51  Collecting google-pasta>=0.1.1
02:47:51    Downloading google_pasta-0.2.0-py3-none-any.whl (57 kB)
02:47:51  Collecting tensorflow-io-gcs-filesystem>=0.21.0
02:47:51    Downloading tensorflow_io_gcs_filesystem-0.21.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.1 MB)
02:47:51  Requirement already satisfied, skipping upgrade: h5py>=2.9.0 in /usr/local/lib/python3.8/dist-packages (from tensorflow==2.8.0) (3.4.0)
02:47:51  Requirement already satisfied, skipping upgrade: keras-preprocessing>=1.1.1 in /usr/local/lib/python3.8/dist-packages (from tensorflow==2.8.0) (1.1.2)
02:47:51  Requirement already satisfied, skipping upgrade: six>=1.12.0 in /usr/lib/python3/dist-packages (from tensorflow==2.8.0) (1.14.0)
02:47:51  Collecting wrapt>=1.11.0
02:47:51    Downloading wrapt-1.13.1-cp38-cp38-manylinux2010_x86_64.whl (84 kB)
02:47:51  [91mERROR: Could not find a version that satisfies the requirement keras<2.8,>=2.7 (from tensorflow==2.8.0) (from versions: 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.0.5, 1.0.6, 1.0.7, 1.0.8, 1.1.0, 1.1.1, 1.1.2, 1.2.0, 1.2.1, 1.2.2, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.0.9, 2.1.0, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.3.0, 2.3.1, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.5.0rc0, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0rc3, 2.6.0, 2.7.0rc0)
02:47:51  ERROR: No matching distribution found for keras<2.8,>=2.7 (from tensorflow==2.8.0)

In general, the installation is checking keras<2.8, and >=2.7. Looks like pypi keras is not released with v2.7?
